internal/kubeclient/transport.go

```diff
@@ -6,27 +6,33 @@ import (
 	netutil "k8s.io/apimachinery/pkg/util/net"
 )
 
+const (
+	xKargoUserCredentialHeader = "X-Kargo-User-Credential"
+)
+
 var (
-	_ netutil.RoundTripperWrapper = &authRoundTripper{}
+	_ netutil.RoundTripperWrapper = &credentialHook{}
 )
 
-type authRoundTripper struct {
+type credentialHook struct {
 	rt http.RoundTripper
 }
 
-func newAuthRoundTripper(rt http.RoundTripper) http.RoundTripper {
-	return &authRoundTripper{
+func newAuthorizationHeaderHook(rt http.RoundTripper) http.RoundTripper {
+	return &credentialHook{
 		rt: rt,
 	}
 }
 
-func (rt *authRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
-	if v, ok := AuthCredentialFromContext(req.Context()); ok {
-		req.Header.Set("Authorization", v)
+func (rt *credentialHook) RoundTrip(req *http.Request) (*http.Response, error) {
```

Comment:
Nit: It would be more conventional to use `c` as the receiver name.

Reply:
I think using `h` is better for this case since the receiver is named to `hook`. WDYT?